ruma-push-gateway-api/src/push/send_event_notification/v1.rs

```diff
@@ -0,0 +1,371 @@
+        /// associated pushers. It may not necessarily be the notification in
+        /// the request that failed: it could be that a previous notification to
+        /// the same pushkey failed. May be empty.
+        pub rejected: Vec<String>,
+    }
+}
+
+impl<'a> Request<'a> {
+    /// Creates a new `Request` with the given notification.
+    pub fn new(notification: Notification<'a>) -> Self {
+        Self { notification }
+    }
+}
+
+impl Response {
+    /// Creates a new `Response` with the given list of rejected pushkeys.
+    pub fn new(rejected: Vec<String>) -> Self {
+        Self { rejected }
+    }
+}
+
+/// Type for passing information about a push notification
+#[derive(Clone, Debug, Default, Outgoing, Serialize)]
+#[cfg_attr(not(feature = "unstable-exhaustive-types"), non_exhaustive)]
+pub struct Notification<'a> {
+    /// The Matrix event ID of the event being notified about.
+    ///
+    /// This is required if the notification is about a particular Matrix event.
+    /// It may be omitted for notifications that only contain updated badge
+    /// counts. This ID can and should be used to detect duplicate notification
+    /// requests.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub event_id: Option<&'a EventId>,
+
+    /// The ID of the room in which this event occurred.
+    ///
+    /// Required if the notification relates to a specific Matrix event.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub room_id: Option<&'a RoomId>,
+
+    /// The type of the event as in the event's `type` field.
+    #[serde(rename = "type", skip_serializing_if = "Option::is_none")]
+    pub event_type: Option<&'a EventType>,
+
+    /// The sender of the event as in the corresponding event field.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub sender: Option<&'a UserId>,
+
+    /// The current display name of the sender in the room in which the event
+    /// occurred.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub sender_display_name: Option<&'a str>,
+
+    /// The name of the room in which the event occurred.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub room_name: Option<&'a str>,
+
+    /// An alias to display for the room in which the event occurred.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub room_alias: Option<&'a RoomAliasId>,
+
+    /// This is `true` if the user receiving the notification is the subject of
+    /// a member event (i.e. the `state_key` of the member event is equal to the
+    /// user's Matrix ID).
+    #[serde(default, skip_serializing_if = "ruma_serde::is_default")]
+    pub user_is_target: bool,
+
+    /// The priority of the notification.
+    ///
+    /// If omitted, `high` is assumed. This may be used by push gateways to
+    /// deliver less time-sensitive notifications in a way that will preserve
+    /// battery power on mobile devices. One of: ["high", "low"]
```

Comment:
```suggestion
    /// battery power on mobile devices.
```

The possible values are contained in the enum.